src/basic/virt.c

```diff
@@ -273,8 +273,8 @@ static Virtualization detect_vm_dmi(void) {
                                                 " assuming virtualized: %m");
                                 return VIRTUALIZATION_AMAZON;
                         }
-                        if (endswith(truncate_nl(s), ".metal")) {
-                                log_debug("DMI product name ends with '.metal', assuming no virtualization");
+                        if (strstr(s, ".metal")) {
```

Comment:
any reason to drop the newline truncation?

Reply:
you don't need the truncation if you don't care if the string has to appear at the end of the string...